### PR TITLE
[@paypal/paypalhttp] Add type definitions

### PR DESCRIPTION
**Pull request by martin-badin**
Discussion: #57699
Documentation: https://github.com/paypal/paypalhttp_node

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test paypal__paypalhttp`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/paypal__paypalhttp/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6",
+            "dom"
```
Comment (peterblazejewicz) [inline types/paypal__paypalhttp/tsconfig.json:6]:
```suggestion
            "es6"
```
why do you need `dom` here?

File: types/paypal__paypalhttp/lib/paypalhttp/http_client.d.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import { Multipart } from './serializer/multipart';
+
+export namespace HttpClient {
+    type Injector = (request: Request) => void;
+
+    interface Headers extends IncomingHttpHeaders {
+        [key: string]: string | string[] | undefined;
+    }
+
+    interface Request<T = object> {
```
Comment (peterblazejewicz) [inline types/paypal__paypalhttp/lib/paypalhttp/http_client.d.ts:16]:
```ts
    interface Request<T = object> {
```
docs explicitly name this `HttpRequest`

File: types/paypal__paypalhttp/lib/paypalhttp/http_client.d.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { IncomingHttpHeaders, ServerResponse } from 'http';
+import { Encoder } from './encoder';
+import { Environment } from './environment';
+import { FormEncoded } from './serializer/form_encoded';
+import { Json } from './serializer/json';
+import { Text } from './serializer/text';
+import { Multipart } from './serializer/multipart';
+
+export namespace HttpClient {
```
Comment (peterblazejewicz) [inline types/paypal__paypalhttp/lib/paypalhttp/http_client.d.ts:9]: do you need a namespace here? with moduel you can control what to export as type (e..g `HttpRequest`)

File: types/paypal__paypalhttp/tsconfig.json (hunk 1 above)
Comment (martin-badin) [reply types/paypal__paypalhttp/tsconfig.json:6]: I am using console in the tests but I will remove it.

File: types/paypal__paypalhttp/lib/paypalhttp/http_client.d.ts (hunk 3 above)
Comment (martin-badin) [reply types/paypal__paypalhttp/lib/paypalhttp/http_client.d.ts:9]: Good idea. I will change it based on you comment

File: types/paypal__paypalhttp/lib/paypalhttp/serializer/multipart.d.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/// <reference types="node" />
+
```
Comment (peterblazejewicz) [inline types/paypal__paypalhttp/lib/paypalhttp/serializer/multipart.d.ts:2]:
```suggestion
```

File: types/paypal__paypalhttp/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+// Type definitions for @paypal/paypalhttp 1.0
+// Project: https://github.com/paypal/paypalhttp_node
+// Definitions by: Martin Badin <https://github.com/martin-badin>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
```
Comment (peterblazejewicz) [inline types/paypal__paypalhttp/index.d.ts:5]:
```suggestion
/// <reference types="node" />

```

File: types/paypal__paypalhttp/lib/paypalhttp/encoder.d.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+/// <reference types="node" />
+
```
Comment (peterblazejewicz) [inline types/paypal__paypalhttp/lib/paypalhttp/encoder.d.ts:2]:
```suggestion
```

File: types/paypal__paypalhttp/test/serializer-tests.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+// }
+
+// {
+//     const multipart = new paypal.Multipart();
+
+//     multipart.contentType(); // $ExpectType RegExp
+//     multipart.encode(req); // $ExpectType Buffer
+//     multipart.decode(); // $ExpectType Error
+//     multipart.formatHeaders(req.headers);
+// }
```
Comment (peterblazejewicz) [inline types/paypal__paypalhttp/test/serializer-tests.ts:50]:
```suggestion
```
no dead code

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@martin-badin thx!

**Comment by martin-badin** [comment]
Ready to merge

**Pull request merged**